academy/xl-cosmjs/intro.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "CosmJs Intro"
+order: 2
+description: What is CosmJs and what can it do for me?
+tag: intro
+---
+
+# The DApp stack
+
+Distributed applications, or DApps, are software applications that run on distributed networks. Blockchains provide persistent data, or state, as well as persistent processes and logic. Cosmos SDK helps developers create such applications. A user interface will be important in most cases. Server interactions are important in many cases. This is where [CosmJs](https://github.com/cosmos/cosmjs) can help.
+
+As the name suggests, CosmJs is a Typescript/Javascript library. It helps developers integrate front-end user interfaces and back-end servers with Cosmos blockchains that implement distributed applications. To many users, the "DApp" _is_ the user interface, even though it's often delivered to the browser in a centralized, traditional way - relying on the DNS infrastructure and centralized web servers.
+
+This re-introduction of a degree of centralization is usually considered acceptable by most teams, provided that the important business logic of the system is enforced by a blockchain and provided it is not strictly necessary to actually use the provided user interface in order to use the DApp. As a general heuristic, offer a user interface as a convenience rather that a necessity.
```

Comment:
Why is the re-introduction acceptable? When is it not? Please elaborate

Reply:
Why offer a UI as a convenience and not a necessity. Please elaborate to be more specific.

Reply:
"Why is the re-introduction acceptable? When is it not? Please elaborate.

Same comment about avoiding too much detail. 

WHY centralization is acceptable -  "the important business logic of the system is enforced by a blockchain and provided it is not strictly necessary to actually use the provided user interface in order to use the DApp"

WHY UI is a convenience - because it is always possible to interact with a blockchain from a command-line interface, forming your own transactions, signing them and usually by creating scripts to do. It's not the preferred method for most users. Users who are concerned about web UIs that originate on centralized servers that might go down, perhaps forever, are reassured that there is _always_ a way to reacquire control of their funds, regardless of what fate befalls the centralized components. 

Each team will design their architecture to their own taste, for their own reasons. We're not trying to be opinionated about it but merely mention the scope of things one can do with CosmJs.